lib/graphql/schema.rb

```diff
@@ -1045,6 +1050,7 @@ def execute(query_str = nil, **kwargs)
         if query_str
           kwargs[:query] = query_str
         end
+        enable_introspection_entry_points if kwargs[:enable_introspection_entry_points]
```

Comment:
Unfortunately, this won't play nice with a Fiber- or Thread-based application server (like Puma) because in that case, multiple queries may be running at the same time, using the same schema. If one query disabled entry points on the schema, then another query enabled them again, the first query would use the setting from the second query. (It's a memory conflict: the schema is shared between queries, so it the schema is changed, all currently-running queries would use the modified schema.) 

Technically, it would still _work_ in forking servers (like Unicorn), but modifying the Schema class while the application is running would cause it to be copied into fork-specific memory which is slow and a poor use of space. (Although I wouldn't be surprised if some other element of Schema setup causes it to be written into the fork's own heap, so that may not be a _new_ problem...)